src/nimblepkg/nimenv.nim

```diff
@@ -0,0 +1,110 @@
+import std/[strscans, os, strutils, strformat]
+import version, nimblesat, cli, common, options
+
+when defined(windows):
+  const
+    BatchFile = """
+  @echo off
+  set PATH="$1";%PATH%
+  """
+else:
+  const
+    ShellFile = "export PATH=$1:$$PATH\n"
+
+const ActivationFile = 
+  when defined(windows): "activate.bat" else: "activate.sh"
+
+proc infoAboutActivation(nimDest, nimVersion: string) =
+  when defined(windows):
+    display("Info", nimDest & "installed; activate with 'nim-" & nimVersion & "activate.bat'")
+  else:
+    display("Info", nimDest & "installed; activate with 'source nim-" & nimVersion & "activate.sh'")
+
+proc compileNim*(options: Options, nimDest: string, v: VersionRange) =
+  let keepCsources = options.useSatSolver #SAT Solver has a cache instead of a temp dir for downloads
+  template exec(command: string) =
+    let cmd = command # eval once
+    if os.execShellCmd(cmd) != 0:
+      display("Error", "Failed to execute: $1" % cmd, Error, HighPriority)
+      return
+  let nimVersion = v.getNimVersion()
+  let workspace = nimDest.parentDir()
+  if dirExists(workspace / nimDest):
+    if not fileExists(nimDest / ActivationFile):
+      display("Info", &"Directory {nimDest} already exists; remove or rename and try again")
+    else:
+      infoAboutActivation nimDest, $nimVersion
+    return
+
+  var major, minor, patch: int
+  if not nimVersion.isSpecial: 
+    if not scanf($nimVersion, "$i.$i.$i", major, minor, patch):
+      display("Error", "cannot parse version requirement", Error)
+      return
+  let csourcesVersion =
+    #TODO We could test special against the special versionn-x branch to get the right csources
+    if nimVersion.isSpecial or (major == 1 and minor >= 9) or major >= 2:
+      # already uses csources_v2
+      "csources_v2"
```

Comment:
More future proof would be:

```nim

if major >= 3: "csources_v" &$major

```

But it's also somewhat risky to predict the future this far.

Reply:
ok but It's an easy enough change to do when `csources` is increased as there is a test already that uses it in the `nimble` ci